core/src/main/java/org/infinispan/marshall/core/internal/ExternallyMarshallable.java

```diff
@@ -0,0 +1,170 @@
+package org.infinispan.marshall.core.internal;
+
+import java.io.Serializable;
+
+// Temporary class, should become active by default when testing
+final class ExternallyMarshallable {
+
+   private ExternallyMarshallable() {
+      // Static class
+   }
+
+   public static boolean isAllowed(Object obj) {
+      String pkg = obj.getClass().getPackage().getName();
+      boolean isBlackList =
+            obj instanceof Serializable
+            && isMarshallablePackage(pkg)
+            && !isWhiteList(obj.getClass().getName());
+      return !isBlackList;
+   }
+
+   private static boolean isMarshallablePackage(String pkg) {
+      return pkg.startsWith("java.")
+            || pkg.startsWith("org.infinispan.")
+            || pkg.startsWith("org.jgroups.")
+            || pkg.startsWith("org.hibernate")
+            || pkg.startsWith("org.apache")
+            || pkg.startsWith("org.jboss")
+            || pkg.startsWith("com.arjuna")
+            ;
+   }
+
+   private static boolean isWhiteList(String className) {
+      return className.endsWith("Exception")
+            || className.contains("$$Lambda$")
+            || className.equals("java.lang.Class")
+            || className.equals("java.util.Date") // test
+            || className.equals("org.infinispan.api.lazy.LazyCacheAPITest$CustomPojo")
```

Comment:
Couldn't we add a marker interface for the test classes, so that only the production classes have to be white-listed here?